### PR TITLE
Expect a `message` property at a minimum to be interpreted as an Error

### PR DESCRIPTION
**Pull request by fregante**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -169,10 +169,23 @@ test('should deserialize array', t => {
...
+});
+
+test('should ignore Error instance', t => {
+	const originalError = new Error('test');
+	const deserialized = deserializeError(originalError);
+	t.is(deserialized, originalError);
+});
+
 test('should deserialize error', t => {
-	const deserialized = deserializeError(new Error('test'));
```
Comment (fregante) [inline test.js:173]: This test was not testing deserialization

File: test.js [hunk 2]
```diff
@@ -169,10 +169,23 @@ test('should deserialize array', t => {
...
+	const deserialized = deserializeError({
+		message: 'Stuff happened',
```
Comment (fregante) [inline test.js:184]: This is the new minimum viable serialized object

File: test.js [hunk 3]
```diff
@@ -169,10 +169,23 @@ test('should deserialize array', t => {
 	deserializeNonError(t, [1]);
 });
 
+test('should deserialize empty object', t => {
+	deserializeNonError(t, {});
```
Comment (fregante) [inline test.js:173]: `serializeError({})` was deserialized as `new Error('')`, now it throws a NonError

File: index.js [hunk 4]
```diff
@@ -78,7 +78,7 @@ const destroyCircular = ({
 			from: value,
 			seen: [...seen],
 
-			to_: isErrorLike(value) ? new Error() : undefined,
+			to_: isMinimumViableSerializedError(value) ? new Error() : undefined,
```
Comment (fregante) [inline index.js:81]: This fixes the inconsistent expectation between the minimum viable error at the top level and for nested errors.

- `{}` was deserialized as an error
- `{cause: {name: 'Error', message: 'foo'}` was serialized as an error, but `cause` wasn't

After #73 this check should also be changed in the new code there.

File: index.js [hunk 5]
```diff
@@ -187,4 +187,11 @@ export function isErrorLike(value) {
 	&& 'stack' in value;
 }
 
+function isMinimumViableSerializedError(value) {
+	return value
+		&& typeof value === 'object'
```
Comment (sindresorhus) [inline index.js:192]:
```suggestion
		&& typeof value === 'object'
		&& value !== null
```

**Comment by sindresorhus** [comment]
I think the docs should explicitly mention what we consider an error when deserializing: https://github.com/sindresorhus/serialize-error#deserializeerrorvalue-options

File: index.js (hunk 5 above)
Comment (fregante) [reply index.js:192]: It already starts with `value` as a truthiness check

Comment (sindresorhus) [reply index.js]: I missed that

**Comment by fregante** [comment]
By the way this is another breaking change, technically, because `{name: "Error"}` would be deserialized as `new Error('')` before while now it's a `new NonError('{"name": "Error"}')`

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Pull request merged**